src/Processor/HotspotProcessor.php

```diff
@@ -7,29 +7,120 @@
 use Intervention\Image\Image;
 use Symfony\Component\HttpFoundation\Request;
 
-final class HotspotProcessor implements Processor
+final readonly class HotspotProcessor implements Processor
 {
+    public function __construct(
+        private ?bool $debug = false,
+    ) {
+    }
+
     public function process(Image $image, Request $request): void
     {
+        $crop = CropProcessor::validateCrop($request);
         if (
             $request->query->has('hotspot')
-            && 1 === preg_match('#^([0-9]+)[x\:]([0-9]+)[x\:]([0-9]+)[x\:]([0-9]+)$#', (string) ($request->query->get('hotspot') ?? ''), $hotspot)
+            && $request->query->has('crop')
+            && ($request->query->has('width') || $request->query->has('height'))
+            && 1 === preg_match('#^(0(?:\.\d+)?|1(?:\.0+)?)[:;x](0(?:\.\d+)?|1(?:\.0+)?)$#', (string) ($request->query->get('hotspot') ?? ''), $hotspot)
+            && 0 < count($crop)
         ) {
-            $pointX = (int) $hotspot[1];
-            $pointY = (int) $hotspot[2];
+            $hotspot = [
+                $hotspot[0],
+                floatval($hotspot[1]),
+                floatval($hotspot[2]),
+            ];
+            // Get width and height with ratio
+            [$width, $height] = $this->getWidthHeight($image, $crop);
+            // Get point X and Y for crop image based on hotspot
+            [$offset_x, $offset_y] = $this->resolveCropOffset($image, $width, $height, $hotspot);
+
+            // Debug mode draw crop and center point on image
+            if ($this->debug) {
```

Comment:
```suggestion
            if ($this->debug && $request->query->has('trace')) {
```

I think you should trigger Tracing in debug mode with a Request query to be able to get the final image in `debug` mode in front of the tracing on original.

- add `trace` query param to Readme **only** in debug mode to see construction lines